api/clients/v2/disperser_client.go

```diff
@@ -95,16 +95,16 @@ func NewDisperserClient(
 ) (*disperserClient, error) {
 
 	if config == nil {
-		return nil, api.NewErrorInvalidArg("config must be provided")
+		return nil, fmt.Errorf("config must be provided")
 	}
 	if config.Hostname == "" {
-		return nil, api.NewErrorInvalidArg("hostname must be provided")
+		return nil, fmt.Errorf("hostname must be provided")
 	}
 	if config.Port == "" {
-		return nil, api.NewErrorInvalidArg("port must be provided")
+		return nil, fmt.Errorf("port must be provided")
 	}
```

Comment:
very nit picky and maybe unnecessary comment but why are we just checking against empty string. wouldn't we also want detection when a multi character empty string is provided; ie:
```
if strings.TrimSpace(val)
```